docker/docker_client.go

```diff
@@ -52,9 +45,13 @@ const (
 	extensionSignatureTypeAtomic    = "atomic" // extensionSignature.Type
 )
 
-// ErrV1NotSupported is returned when we're trying to talk to a
-// docker V1 registry.
-var ErrV1NotSupported = errors.New("can't talk to a V1 docker registry")
+var (
+	// ErrV1NotSupported is returned when we're trying to talk to a
+	// docker V1 registry.
+	ErrV1NotSupported = errors.New("can't talk to a V1 docker registry")
+	// ErrUnauthorized is returned when the status code returned is 401
+	ErrUnauthorized = errors.New("unable to retrieve auth token: 401 unauthorized")
```

Comment:
drop "401 unauthorized", that's an implementation detail we don't want to expose directly in the API

Reply:
I see it was this way in the past, but we should not expose the error, can you make it private? (lowercasing it)

Reply:
… This should probably say something more generic like “invalid credentials”, it is returned also on the `Authorization: Basic` path where no tokens are involved.

Reply:
> I see it was this way in the past, but we should not expose the error,

This should allow callers of `CheckAuth` to distinguish between unexpected failures (e.g. network down) and an invalid username/password.

Reply:
So I can leave the error exposed and just change the error message?

Reply:
I just remembered, I am using this error in kpod login to know when a 401 is returned so that the message printed for the user is along the paths of "invalid username/password", hence making it public.

Reply:
So given the places this error is used it must not contain "401 ..." cause it's HTTP related and should probably be named something like `ErrUnauthorizedForCredentials` (better naming is advised here but just `ErrUnauthorized` is misleading).

Reply:
got it, fixed.